### PR TITLE
Felix: int dataplane main loop cleanups

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Upstreaming some changes form Enterprise.  We needed an "in sync" message from the iface monitor in Enterprise and the code was very messy so it needed a refactor.

- Add an in-sync message to the interface monitor. (Required combining its channels into one for synchronisation).
- Move 'peek' loops out of main loop method and introduce common drainChan() helper.
- Block dataplane updates until interface monitor is in sync.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/dataplane/linux/int_dataplane.go [hunk 1]
```diff
@@ -1871,6 +1780,127 @@ func (d *InternalDataplane) loopUpdatingDataplane() {
...
+		d.processIfaceInSync()
+	}
+}
+
+func (d *InternalDataplane) processIfaceInSync() {
+	if d.ifaceMonitorInSync {
+		return
+	}
+	log.Info("Interface monitor now in sync.")
+	d.ifaceMonitorInSync = true
```
Comment (tomastigera) [inline felix/dataplane/linux/int_dataplane.go:1860]: does this ever become false? Seems like the iface monitor may set it many times to true

**Review by tomastigera — COMMENTED** [review]
LGTM

File: felix/dataplane/linux/int_dataplane.go (hunk 1 above)
Comment (fasaxc) [reply felix/dataplane/linux/int_dataplane.go:1860]: No, it doesn't; left over from previous iteration.

Comment (fasaxc) [reply felix/dataplane/linux/int_dataplane.go:1860]: No, it's a one-way flag; just to let the main loop know that we're in sync at one point.

**Pull request merged**